### DISCUSSION TITLE
Does Avalonia support Visual Studio's Live Unit Testing?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14741] [Q&A]
When I turn Live Unit Testing on in Visual Studio, all Avalonia-related methods and variables show a CS0103 error, indicating that the name does not exist in the current context. However, the program can still run successfully. Is there any setting that I missed?

![2024-02-27 101120](https://github.com/AvaloniaUI/Avalonia/assets/9688223/95ad7b3a-d45e-4873-ab2b-94295dca6c79)

**Comment by maxkatz6** [comment]
It might be that Live Unit Testing doesn't support C# Source Generators. Probably unrelated to Avalonia specifically.

**Reply by IronEaterBeast** [reply]
Thank you. So, there is no way to solve this issue through any settings?

**Reply by maxkatz6** [reply]
No, I don't know. Visual Studio support might be able to help here.

**Comment by thevortexcloud** [comment]
You should be able to unit test view models without issue. If you are using ReactiveUI there is some additional setup required though.

https://www.reactiveui.net/docs/handbook/testing.html
